test/sequential/test-cluster-disconnect-leak.js

```diff
@@ -1,46 +1,21 @@
 'use strict';
-// Flags: --expose-internals
+
+// Test fails in Node v5.4.0 and passes in v5.4.1 and newer.
 
 const common = require('../common');
-const assert = require('assert');
 const net = require('net');
 const cluster = require('cluster');
-const handles = require('internal/cluster').handles;
-const os = require('os');
-
-if (common.isWindows) {
-  console.log('1..0 # Skipped: This test does not apply to Windows.');
-  return;
-}
 
 cluster.schedulingPolicy = cluster.SCHED_NONE;
 
 if (cluster.isMaster) {
-  const cpus = os.cpus().length;
-  const tries = cpus > 8 ? 128 : cpus * 16;
-
-  const worker1 = cluster.fork();
-  worker1.on('message', common.mustCall(() => {
-    worker1.disconnect();
-    for (let i = 0; i < tries; ++ i) {
-      const w = cluster.fork();
-      w.on('online', common.mustCall(w.disconnect));
-    }
-  }));
-
-  cluster.on('exit', common.mustCall((worker, code) => {
-    assert.strictEqual(code, 0, 'worker exited with error');
-  }, tries + 1));
-
-  process.on('exit', () => {
-    assert.deepEqual(Object.keys(cluster.workers), []);
-    assert.strictEqual(Object.keys(handles).length, 0);
-  });
-
+  // This is the important part of the test: Confirm that `disconnect` fires.
+  cluster.fork().on('disconnect', common.mustCall(() => {}));
+  cluster.disconnect();
```

Comment:
We could also check that the `exit` event is called on the worker and that the `cluster.disconnect` callback is also called